model-engine/model_engine_server/infra/gateways/gcs_file_storage_gateway.py

```diff
@@ -0,0 +1,104 @@
+import os
+from typing import List, Optional
+
+from google.cloud import storage
+
+from model_engine_server.core.config import infra_config
+from model_engine_server.domain.gateways.file_storage_gateway import (
+    FileMetadata,
+    FileStorageGateway,
+)
+from model_engine_server.infra.gateways.gcs_filesystem_gateway import GCSFilesystemGateway
+
+
+def get_gcs_key(owner: str, file_id: str) -> str:
+    """
+    Constructs a GCS object key from the owner and file_id.
+    """
+    return os.path.join(owner, file_id)
+
+
+def get_gcs_url(owner: str, file_id: str) -> str:
+    """
+    Returns the gs:// URL for the bucket, using the GCS key.
+    """
+    return f"gs://{infra_config().gcs_bucket}/{get_gcs_key(owner, file_id)}"
+
+
+class GCSFileStorageGateway(FileStorageGateway):
+    """
+    Concrete implementation of a file storage gateway backed by GCS.
+    """
+
+    def __init__(self):
+        self.filesystem_gateway = GCSFilesystemGateway()
+
+    async def get_url_from_id(self, owner: str, file_id: str) -> Optional[str]:
+        """
+        Returns a signed GCS URL for the given file.
+        """
+        try:
+            return self.filesystem_gateway.generate_signed_url(get_gcs_url(owner, file_id))
+        except Exception:
+            return None
+
+    async def get_file(self, owner: str, file_id: str) -> Optional[FileMetadata]:
+        """
+        Retrieves file metadata if it exists. Returns None if the file is missing.
+        """
+        try:
+            client = self.filesystem_gateway.get_storage_client({})
+            bucket = client.bucket(infra_config().gcs_bucket)
```

Comment:
I know this pattern was already there, but I think it'd probably make more sense to pass in the bucket into the constructor of this class. This way, there's one less dependency on the old `infra_config` object. @seanshi-scale @tiffzhao5 thoughts?

Could also make the argument to just pass in the bucket as an argument with every `get_file` call, but that's outside of the scope of this change I'd say.

Reply:
I'm fine with having the bucket be passed in the constructor (in addition to anything else from any configs); `dependencies.py` does read in from infra_config at times to figure out constructor arguments, so there's precedent already